### PR TITLE
Control `MemoryDiagnoser` benchmark option via CLI instead of code

### PR DESCRIPTION
**Pull request by mic-max**
## Changes

The idea is to remove all the `MemoryDiagnoser` class attributes that we include in all the benchmark classes and instead just add the console argument -m to always enable it.

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by mic-max** [comment]
If people think this change is worthwhile I can do a final commit that removes all `[MemoryDiagnoser]` lines from benchmark classes like done here https://github.com/open-telemetry/opentelemetry-dotnet/pull/3085/commits/a57afff61448084329cc9183e9ff68ead900dcdc

File: test/Benchmarks/EventSourceBenchmarks.cs [hunk 1]
```diff
@@ -20,7 +20,6 @@
 
 namespace OpenTelemetry.Benchmarks
 {
-    [MemoryDiagnoser]
```
Comment (cijothomas) [inline test/Benchmarks/EventSourceBenchmarks.cs:23]: what is the benefit of providing this via command line? 
The cmd line is straightforward right now, as one can run it just like any dotnet run. I don't see any advantage of making the command line user needing to remember one more param.

**Review by reyang — APPROVED** [review]

File: test/Benchmarks/EventSourceBenchmarks.cs (hunk 1 above)
Comment (reyang) [reply test/Benchmarks/EventSourceBenchmarks.cs:23]: E.g. if I made a change that I know won't change any memory alloc, I could run the test in a much faster way 😃

Comment (cijothomas) [reply test/Benchmarks/EventSourceBenchmarks.cs:23]: okay that seems to be a good reason.

File: test/Benchmarks/README.md [hunk 2]
```diff
@@ -1,10 +1,7 @@
 # OpenTelemetry Benchmarks
 
-Use the following example to run Benchmarks from command line:
-(change parameters as necessary)
-
 Navigate to `./test/Benchmarks` directory and run the following command:
 
 ```sh
-dotnet run --framework net6.0 --configuration Release --filter *TraceBenchmarks*
+dotnet run -c Release -f net6.0 -- -m
```
Comment (cijothomas) [inline test/Benchmarks/README.md:6]: if we remove the filter , please add example with the filter as well below (to make it easy to copy paste to cmd).
Also - add a one line explaining why would folks want to use/not-use -m parameter.

**Review by reyang — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]

**Pull request merged**